.github/actions/e2e/run-log-tests/action.yml

```
@@ -27,9 +32,32 @@ runs:
     - name: Re-try Failed E2E Files
       if: ${{ steps.first_run_e2e_tests.outputs.FIRST_RUN_FAILED_TEST_SUITES > 0  }}
       shell: bash
-      # Filter failed E2E files from the result JSON file, and re-run them.
       run: |
-        npm run test:e2e-ci $(cat $E2E_RESULT_FILEPATH | jq -r '[.suites[] | (if has("suites") then .suites[] | .specs[] else .specs[] end) | select(.tests[].status == "unexpected") | .file] | unique | .[]')
+        set -e
+        RESULTS_JSON="${{ steps.first_run_e2e_tests.outputs.RESULTS_JSON }}"
+        echo "Using results file: $RESULTS_JSON"
+
+        # Build a unique list of spec files that had unexpected/failed outcomes.
+        # This uses a recursive jq search to find any node with a 'specs' array, then selects specs
+        # where any test result is marked unexpected/failed.
+        mapfile -t FAILED_SPECS < <( jq -r '[ .. | objects | select(has("specs")) | .specs[]
+          | select( ( any(.tests[]?; (.status=="unexpected") or (.status=="failed")) )
+            or ( any(.tests[]?.results[]?; (.status=="failed") or (.status=="timedOut") or (.status=="interrupted")) ) )
+          | .file ] | unique | .[]' "$RESULTS_JSON" )
+
+        if [[ ${#FAILED_SPECS[@]} -eq 0 ]]; then
+          echo "::notice::No failed specs found in results file. Re-running full suite instead."
+          npm run test:e2e-ci
+          exit 0
+        fi
+
+        echo "Retrying failed specs (${#FAILED_SPECS[@]}):"
+        for f in "${FAILED_SPECS[@]}"; do
+          echo " - $f"
+        done
+
+        # Re-run only the failed spec files using Playwright directly (npm script does not accept args)
+        npx playwright test --config=tests/e2e/playwright.config.ts --grep-invert @todo "${FAILED_SPECS[@]}"
```

I think this should work 🤔 
```suggestion
        # Re-run only the failed spec files. 
        npm run test:e2e-ci "${FAILED_SPECS[@]}"
```

---

Indeed, done here [97b6efd](https://github.com/Automattic/woocommerce-payments/pull/11035/commits/97b6efd5a16db885a4b8e2a4a6a906c12e2afdb6)